btp-document/Smart-Contracts-ICON.md

```diff
@@ -0,0 +1,387 @@
+
+&emsp;   `-PkeystorePass=`: a password, that is saved in a `keysecret` file, which is generated at the same time with `keystore.json`
+
+For example: 
+
+```shell
+./gradlew deployToLocal -PkeystoreName="/Users/myfolder/INode/keystore.json" -PkeystorePass=5a17fdee471e5dd4
+```
+
+### 7. ICON-BMC Configuration
+
+____
+
+**Note that: This configuration step must be executed after completely deploying smart contracts (BSH, BMC, and BMV) on both connecting networks. In this example, we would like to show how to config to setup a connection between ICON and Moonriver networks. For other cases, there might be a different configuration**
+
+#### Add **BMV**
+
+**TBA**
+
+```bash
+
+```
+
+#### Add Connection Link to Moonriver-BMC
+
+&emsp; This step requires Moonriver chainID, and an address of Moonriver-BMC contract. The information is generated after deploying Moonbeam-BMC contract ([link](Smart-Contracts-PRA.md#deploy-bmc-contracts-on-moonriver-network))
+
+```shell
+echo "btp://${Moonriver_CHAINID}.pra/${MoonriverBMC}" > btp.dst
+# For example: Moonriver_CHAINID = 0x501; MoonriverBMC = 0x5CC307268a1393AB9A764A20DACE848AB8275c46
+
+rpcch icon
+goloop rpc sendtx call --to $(cat bmc.$(rpcch)) \
+    --step_limit 13610920001 \
+    --method addLink \
+    --param _link=$(cat btp.dst) \
+    | jq -r . > tx.link.$(rpcch)
+```
+
+In success, a connection link from ICON-BMC to Moonriver-BMC will be set with default setting values:
+
+&emsp; &emsp; &emsp; + `BLOCK_INTERVAL_MSEC`: block interval of Moonriver (default = 1000ms)
+
+&emsp; &emsp; &emsp; + `MAX_AGGREGATION`: max_aggregation value (default = 10)
+
+&emsp; &emsp; &emsp; + `DELAY_LIMIT`: acceptance of delayed submission sending from BMR to BMC (default = 3)
+
+<p align="center">
+  <img src="./images/bmc_rotate_round-robin.png" width="800" height="300" />
+</p>
+
+<p align="center">
+  <img src="./images/bmc_rotate_fail-over.png" width="800" height="300" />
+</p>
+
+#### Set Link Configuration
+
+&emsp; This step will help you to change default setting values
+
+```bash
+rpcch icon
+goloop rpc sendtx call --to $(cat bmc.$(rpcch)) \
+    --step_limit 13610920001 \
+    --method setLink \
+    --param _link=$(cat btp.dst) \
+    --param _block_interval=0x1770 \
+    --param _max_agg=0x08 \
+    --param _delay_limit=4 \
+    | jq -r . > tx.setlink.$(rpcch)
+```
+
+#### Add BSH Service contract
+
+```bash
+rpcch icon
+goloop rpc sendtx call --to $(cat bmc.$(rpcch)) \
+    --step_limit 13610920001 \
+    --method addService \
+    --param _addr=$(cat nativeCoinBsh.$(rpcch)) \
+    --param _svc=CoinTransfer \
+    | jq -r . > tx.addService.$(rpcch)
+```
+
+#### Register Relay to ICON-BMC
+
+&emsp; **Note that: To complete this step, keystore of BMR from Moonriver --> ICON ('icon-bmr.keystore.json') must be generated.** Please check out this [link](BMR-deployment.md#create-keystore-files) to complete this requirement
```

Comment:
Broken link 404

Reply:
Maybe https://github.com/quangtran8588/devportal/blob/wip/btp-document/btp-document/BMR-Deployment.md#2-generate-keystore-and-configuration-files ?